### ISSUE TITLE
Doesn't seem to work any more after recent Wercker changes

### ISSUE THREAD

**Issue by pimterry** [wercker/step-grunt#8]
I'm not sure exactly where to put this, but I think here seems best. I have a grunt deploy step (which just runs `grunt release`), and it's suddenly stopped working. Outline of the trace is:

```bash
export WERCKER_STEP_ROOT="/pipeline/grunt-19a18dd8-5391-4b91-951d-1822fc80237b"
export WERCKER_STEP_ID="grunt-19a18dd8-5391-4b91-951d-1822fc80237b"
export WERCKER_STEP_OWNER="wercker"
export WERCKER_STEP_NAME="grunt"
export WERCKER_REPORT_NUMBERS_FILE="/report/grunt-19a18dd8-5391-4b91-951d-1822fc80237b/numbers.ini"
export WERCKER_REPORT_MESSAGE_FILE="/report/grunt-19a18dd8-5391-4b91-951d-1822fc80237b/message.txt"
export WERCKER_REPORT_ARTIFACTS_DIR="/report/grunt-19a18dd8-5391-4b91-951d-1822fc80237b/artifacts"
export WERCKER_GRUNT_TASKS="release"
source "/pipeline/grunt-19a18dd8-5391-4b91-951d-1822fc80237b/run.sh" < /dev/null
grunt-cli not installed, trying to install it through npm
npm is available
npm version: 3.3.12
installing grunt-cli
- abbrev@1.0.7 node_modules/npm/node_modules/abbrev
- ansi-regex@2.0.0 node_modules/npm/node_modules/ansi-regex

[...lots and lots of packages later...]

- async-some@1.0.2 node_modules/npm/node_modules/async-some
- write-file-atomic@1.1.3 node_modules/npm/node_modules/write-file-atomic
/pipeline/grunt-19a18dd8-5391-4b91-951d-1822fc80237b/run.sh: line 30: /usr/local/bin/npm: No such file or directory
grunt release
/pipeline/grunt-19a18dd8-5391-4b91-951d-1822fc80237b/run.sh: line 70: grunt: command not found
grunt exited with exit code: 127
failed: grunt failed
```

Full deploy details are available at https://app.wercker.com/#deploystep/5655ef008faa27fb0d12a025. Previous completely successful deploy of this project was on the 13th and is up at https://app.wercker.com/#deploy/56463ab0817267310c00d620

This has been working for months, I haven't changed anything about it recently, and I think it's fairly clear that it's failing in the initial Grunt setup. Notably the log above is quite different to what I've been seeing before, which used to just say:

```bash
export WERCKER_STEP_ROOT="/pipeline/grunt-0e43682a-f44b-40b5-ae26-4d10d7b3ed46"
export WERCKER_STEP_ID="grunt-0e43682a-f44b-40b5-ae26-4d10d7b3ed46"
export WERCKER_STEP_OWNER="wercker"
export WERCKER_STEP_NAME="grunt"
export WERCKER_REPORT_NUMBERS_FILE="/report/grunt-0e43682a-f44b-40b5-ae26-4d10d7b3ed46/numbers.ini"
export WERCKER_REPORT_MESSAGE_FILE="/report/grunt-0e43682a-f44b-40b5-ae26-4d10d7b3ed46/message.txt"
export WERCKER_REPORT_ARTIFACTS_DIR="/report/grunt-0e43682a-f44b-40b5-ae26-4d10d7b3ed46/artifacts"
export WERCKER_GRUNT_TASKS="release"
source "/pipeline/grunt-0e43682a-f44b-40b5-ae26-4d10d7b3ed46/run.sh" < /dev/null
grunt-cli not installed, trying to install it through npm
npm is available
npm version: 2.11.3
installing grunt-cli
/usr/local/bin/npm -> /usr/local/lib/node_modules/npm/bin/npm-cli.js
npm@3.4.0 /usr/local/lib/node_modules/npm
/usr/local/bin/grunt -> /usr/local/lib/node_modules/grunt-cli/bin/grunt
/usr/local/lib
`-- grunt-cli@0.1.13 
  +-- findup-sync@0.1.3 
  | +-- glob@3.2.11 
  | | +-- inherits@2.0.1 
  | | `-- minimatch@0.3.0 
  | |   +-- lru-cache@2.7.0 
  | |   `-- sigmund@1.0.1 
  | `-- lodash@2.4.2 
  +-- nopt@1.0.10 
  | `-- abbrev@1.0.7 
  `-- resolve@0.3.1 

grunt release
Running "webstore_upload" task

[... actual build process ...]
```

Any ideas?

**Comment by hatchan** [comment]
@pimterry For some reason this step will update npm to the latest version :-/ Which somehow doesn't work as expected anymore. 

Also the logs for the installation of npm and grunt-cli are suppressed, so can't check if something went wrong, and what went wrong.

**Comment by hatchan** [comment]
Added #9 and #10. Will look into this later today or tomorrow.

**Comment by pimterry** [comment]
This same issue is now also breaking all my normal builds, not just deploys: https://app.wercker.com/#buildstep/5658861d5d4798100d08fe9b

Maybe it's just that the bug appeared changed and I've only run the deploy since, but not a new build, or maybe this is a new development, not clear. Still, my build process is all totally broken in the meantime until this gets fixed :-(

Anything I can do to help?

**Comment by hatchan** [comment]
I've just created a pull request #11. I am leaving in a couple of minutes, so I'm not super confident in releasing this right now.

**Comment by pimterry** [comment]
Ha, fair enough! Sensible. Waiting until the start of next week is fine.

**Comment by hatchan** [comment]
Released `2.0.0` of this step. Could you check if the problem still occurs, and if it does, do the logs contain some useful information?

**Comment by pimterry** [comment]
Thanks @hatchan. This has fixed the issue with Grunt for this build very nicely, good stuff!

I am still seeing the bug from [my other issue](https://github.com/wercker/support/issues/168) (both store-container not working and my ftp deploy script has suddenly mysteriously failing), which I was hoping would be related to some NPM or Docker changes, but sounds like no such luck. If you've got any clues for things you might've changed that would be useful, but otherwise I'll keep digging in.

This issue's fixed for now anyway, thanks for your help!
